### DISCUSSION TITLE
Find a span

### DISCUSSION THREAD

**Discussion by kvor** [open-telemetry/opentelemetry-java#3652] [Q&A]
Is it possible to find a span (within a Trace)  by  name or even by Id?

**Comment by jkwatson** [comment]
Can you explain a little bit more about what you need? Where are you looking for spans, exactly?

**Comment by kvor** [comment]
@jkwatson Thanks for your reply. What we want is to close a span within a trace. Since the code is generic, we do not have explicit calls to start/end a span case by case inside a specific block of code but rather generic ones (for example start/end spans inside a filter). However, we do want to close all opened spans when an exception happens.

**Reply by jkwatson** [reply]
There's definitely no built-in way to do this. How are the spans getting created?
